File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/Java/C++/s867221485_commented.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
    int N;
    string s;
    bool first = true;
    int ans = 0;
    int sha = 0;
    int dot = 0;

    cin >> N;
    cin.ignore();
    getline(cin, s);

    for (int i = 0; i < N; i++) {
        char c = s[i];
        if (c == '#') {
            first = false;
            if (sha <= dot) {
                ans += sha;
                sha = 0;
                dot = 0;
                first = true;
            }
            sha++;
        } else {
            if (!first) {
                dot++;
            }
        }
    }

    if (sha <= dot) {
        ans += sha;
        dot = 0;
    }

    cout << ans + dot << endl;
    return 0;
}
// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/mad/Java/Go/s867221485_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	N, _ := strconv.Atoi(scanner.Text())

	scanner.Scan()
	s := scanner.Text()

	ans := 0
	sha := 0
	dot := 0
	first := true
	for i := 0; i < N; i++ {
		if s[i] == '#' {
			first = false
			if sha <= dot {
				ans += sha
				sha = 0
				dot = 0
				first = true
			}
			sha++
		} else {
			if !first {
				dot++
			}
		}
	}

	if sha <= dot {
		ans += sha
	}

	fmt.Println(ans + dot)
}

// End of Code
